screen2.0/src/app/search/MainResultsFilters.tsx

```diff
@@ -0,0 +1,767 @@
+"use client"
+import * as React from "react"
+import {
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+  Typography,
+  Button,
+  Paper,
+  FormGroup,
+  FormControlLabel,
+  Checkbox,
+  TextField,
+  Tooltip,
+  Box,
+  Slider,
+  FormLabel,
+  Stack,
+  CircularProgress,
+} from "@mui/material/"
+
+import Radio from '@mui/material/Radio';
+import RadioGroup from '@mui/material/RadioGroup';
+import FormControl from '@mui/material/FormControl';
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore"
+import KeyboardArrowRightIcon from "@mui/icons-material/KeyboardArrowRight"
+import Grid2 from "@mui/material/Unstable_Grid2"
+import { RangeSlider, DataTable } from "@weng-lab/psychscreen-ui-components"
+import { useState, useMemo, useEffect } from "react"
+import { BiosampleTableFilters, CellTypeData, FilterCriteria, FilteredBiosampleData, MainQueryParams } from "./types"
+import { parseByCellType, filterBiosamples, assayHoverInfo } from "./search-helpers"
+import { gql } from "@apollo/client"
+import { useQuery } from "@apollo/experimental-nextjs-app-support/ssr"
+import GeneAutoComplete from "../applets/gene-expression/gene-autocomplete";
+import { InfoOutlined } from "@mui/icons-material";
+
+const snpMarks = [
+  {
+    value: 0,
+    label: '0kb',
+  },
+  {
+    value: 500,
+    label: '0.5kb',
+  },
+  {
+    value: 1000,
+    label: '1kb',
+  },
+  {
+    value: 1500,
+    label: '1.5kb',
+  },
+  {
+    value: 2000,
+    label: '2kb',
+  }
+];
+
+const tssMarks = [
+  {
+    value: 0,
+    label: '0kb',
+  },
+  {
+    value: 10000,
+    label: '10kb',
+  },
+  {
+    value: 25000,
+    label: '25kb',
+  },
+  {
+    value: 50000,
+    label: '50kb',
+  },
+  {
+    value: 100000,
+    label: '100kb',
+  }
+];
+
+const GENE_TRANSCRIPTS_QUERY = gql`
+ query ($assembly: String!, $name: [String!], $limit: Int) {
+   gene(assembly: $assembly, name: $name, limit: $limit) {
+     name
+     id
+     coordinates {
+       start
+       chromosome
+       end
+     }
+     strand
+     transcripts {
+      name
+      coordinates {
+        start
+        end
+      }      
+    }
+   }
+ } `
+
+ //TODO
+ // gene changing main query
+
+export default function MainResultsFilters(
+  props: {
+    mainQueryParams: MainQueryParams,
+    setMainQueryParams: React.Dispatch<React.SetStateAction<MainQueryParams>>,
```

Comment:
If React imported globally in file (import React from "react"), we can get rid f React at multiple place.

Reply:
Need to import Dispatch separately, but did so to clean this up